### ISSUE TITLE
EVAL runSnippet Not Working

### ISSUE THREAD

**Issue by rossco** [modxbot/migrate#4616]
rossco created Redmine issue ID 4616

After upgrading from rc3 to rc4 (revo 2.1) I have a snippet that returns a basic list that TVs use to create a drop down list from.  

In my TV I use the Input Option Values:   @EVAL return $modx->runSnippet('getForms');  

The getForms snippet returns the following when used alone in the MODx document.  Contact Us==1||Services==2

In rc3 this runs fine, rc4 it will not fire the snippet.  No error is logged in the MODx manager.

**Comment by rossco** [comment]
rossco submitted:

Some extra information.

I decided to do a complete fresh install of rc4 and a new DB but this has not helped.   Changing the snippet to simply return 'options1==1||options2==2'; has not helped.  Again the snippet works fine standalone but it will not work using @EVAL.

When going to a document that the TV is available on shows this error.

Parse error: syntax error, unexpected $end in /home/mysite/public_html/dev/core/model/modx/modtemplatevar.class.php(693) : eval()'d code on line 1

eval is enabled in PHP.  Same server, using rc3, same snippet, same TV setup works.  :D

**Comment by opengeek** [comment]
opengeek submitted:

I am unable to reproduce this issue. From the error message you are reporting, there seems to be a problem with your Snippet syntax. Can you double-check the getForm snippet or provide additional information so we can reproduce?

**Comment by rossco** [comment]
rossco submitted:

Hi Jason,

the snippet is working now in the TV, I have not touched it.  I did clear the cache via FTP when it was not working and signed in and out, uploaded and upgraded to rc4 3 times.  I left the site for 2 days and came back, it's now working.  I have no idea what was wrong but it's now working.

I don't think a browser would have any effect on this but it's working.  The only thing that I did do after a day of it not working is done a fresh install of RC4 rather then rc3-rc4 upgrade.  Perhaps some corruption in the files during upgrade?  I do not have a copy of rc3 before the upgrade so on this occasion I cannot see if I can hit the issue again.  

Hopefully this is just a corrupt upload file or something silly.  Thanks!

---- Copy of getForms snippet ----

The getForm snippet gets the ID and Name of a form created in an external application that is on the same server.  It's called Machform and has it's own database.  This is why I had to connect using the following. 

<pre>
<?php
$dsn = 'mysql:host=localhost;dbname=forms;port=3306;charset=utf-8';
$xpdo = new xPDO($dsn,'usr','pw');
  //  echo $o=($xpdo->connect()) ? 'Connected' : 'Not Connected';
$stmt =  $xpdo->query("SELECT form_id,form_name FROM ap_forms WHERE form_active = 1");
if ($stmt) {
          while ($row = $stmt->fetch(PDO::FETCH_ASSOC)) {
              $children[] = array(
                'title' => $row['form_name'],
                'id' => $row['form_id'],
              );            
           }   
        
  $resourceIds = "";
  $i=0;
  
  foreach ($children as $child) {
     if($i!=0){
        $resourceIds .= "||";
     }
    
     //$resourceIds .= $val;
    
     $resourceIds .= $child['title'];
     $resourceIds .= "==";
     $resourceIds .= $child['id'];
    
     $i++;
  }
  return $resourceIds;
  
  }
</pre>
